src/backend/drm/device/mod.rs

```diff
@@ -406,6 +406,19 @@ impl DrmDevice {
                 error!("Failed to acquire drm master again. Error: {}", err);
             }
         }
+        // FIXME: This is not ideal, but we clear all crtcs here to prevent issues later when trying to
+        // commit a surface. The reason is that after a tty switch (which is to most likely cause for a session resume
+        // and calling this function) we might end up in a situation where the CRTC <-> Connector relationship is no
+        // longer like we left it before the session pause. The worst state is where the two CRTC got swapped. In this
+        // case we fail to update the state on the next commit because we currently always only update a single CRTC.
+        // This won't work because our CRTC is still bound to the other connector.
+        //
+        // I am not sure how we can solve this, but for now just clearing all CRTCs on a session resume should reliable solve this issue.
+        for crtc in self.crtcs() {
+            if let Err(err) = self.set_crtc(*crtc, None, (0, 0), &[], None) {
+                tracing::warn!(?err, ?crtc, "failed to reset crtc on active");
+            }
+        }
```

Comment:
This looks a lot like the `reset_state` method we already have for our `AtomicDrmDevice` and `LegacyDrmDevice`, which are used for exactly that purpose when initializing the `DrmDevice` with `disable_connectors` set to `true`.

Can we re-use these here please?

Reply:
wdym? reset_state gets called on session resume in anvil, but currently only reads the current state from drm. The actual change is done in commit, which faces exactly the described issue. Also reset_state is per surface, so this would rely on downstream call reset_state on all surfaces "atomically". Also it might not solve the issue in all cases I believe.

Reply:
Ah, you are talking about the device, not the surface. Yeah, sure, sorry.

Reply:
Hm, thinking about the whole thing again it might be better to make that explicit. Maybe by adding a disable_connectors parameter to activate which resets the state like in new. We could also make reset_state public, but I feel like that alone wouldn't be enough to make downstream aware of this issue. So maybe we should do both, add the parameter and break existing code making it aware of the issue and let downstream decide how to handle it.